### PR TITLE
Limit "releases" in history (compact view) not items in history

### PR DESCRIPTION
**Pull request by fernandog**
@duramato @MGaetan89 

Before:
image

After:
image

### CODE REVIEW

**Comment by fernandog** [comment]
@MGaetan89 i don't see any impact in the API. is this ok for you ?

**Comment by MGaetan89** [comment]
I don't think it is working.
Because in the API, we do `data = History().get(self.limit, self.type)`, but the limit parameter no longer exists (https://github.com/pymedusa/SickRage/blob/master/sickbeard/webapi.py#L1050).
It also mean that the `limit` parameter of the API does nothing now, right?

**Comment by fernandog** [comment]
@MGaetan89 

to work in the API we should move this code: https://github.com/pymedusa/SickRage/blob/history_fix/sickbeard/webserve.py#L3888-L3914

to sickrage/show/History.py
so we add back the limit to History.py 

what do you think?:

**Comment by fernandog** [comment]
@MGaetan89 i'm getting this with this last commit:

```
  File "/home/osmc/SickRage/sickrage/show/History.py", line 109, in get
    history['actions'].append(available_actions)
KeyError: 'actions'

```

File: sickrage/show/History.py [hunk 1]
```diff
@@ -60,7 +62,7 @@ def get(self, action=None):
         else:
             results = self.db.select(common_sql + order_sql)
 
-        data = []
+        data = compact = []
```
Comment (labrys) [inline sickrage/show/History.py]: this should be two separate lines for clarity.

File: sickrage/show/History.py [hunk 2]
```diff
@@ -74,7 +76,40 @@ def get(self, action=None):
                 'show_name': result['show_name']
             })
```
Comment (labrys) [inline sickrage/show/History.py]: This should be a list comprehension instead:
```python
data = [{'action': result['action'],
         'date': result['date'],
         'episode': result['episode'],
         'provider': result['provider'],
         'quality': result['quality'],
         'resource': result['resource'],
         'season': result['season'],
         'show_id': result['showid'],
         'show_name': result['show_name'],
         } for result in results]
```

File: sickrage/show/History.py [hunk 3]
```diff
@@ -74,7 +76,40 @@ def get(self, action=None):
...
+                'action': row['action'],
+                'provider': row['provider'],
+                'resource': row['resource'],
+                'time': row['date']
+            }
+
+            if not any((history['show_id'] == row['show_id'] and
+                        history['season'] == row['season'] and
+                        history['episode'] == row['episode'] and
+                        history['quality'] == row['quality']) for history in compact):
```
Comment (labrys) [inline sickrage/show/History.py]: this should be an all within an any.

Comment (fernandog) [inline sickrage/show/History.py]: @labrys 
did all other changes but didn't understand this

Comment (labrys) [inline sickrage/show/History.py]:
```python
if not any(all([history['show_id'] == row['show_id'],
                history['season'] == row['season'],
                history['episode'] == row['episode'],
                history['quality'] == row['quality'],
                ]) for history in compact):
```
or even better yet:
```python
if not any(all(history[arg] == row[arg] for arg in
               ['show_id', 'season', 'episode','quality',])
           for history in compact):
```

**Comment by labrys** [comment]
FYI: When there's line notes it's better to add commits to fix the changes or even PR to the branch to avoid eradicating those notes.

**Comment by p0psicles** [comment]
I'm getting the following mako error:
image

And at time of the error hItem has the following value:
dict: {'show_id': 247808, 'resource': u'Suits.S02E01-E10.1080p.WEB-DL.DD5.1.H.264', 'season': 2, 'show_name': u'Suits', 'actions': [{'action': 6402, 'time': 20160321155657L, 'resource': u'Suits.S02E01-E10.1080p.WEB-DL.DD5.1.H.264', 'provider': u'oznzb'}], 'episode': 2, 'quality': 64}

You see that it's expecting the key 'action', but that doesn't exist anymore. I expect you want to use the 'actions' key. But I don't know if it's possible to pass multiple dicts in that list?

I'm on be4fc41a8ae57216acbd3053768d1f40d380f327

File: sickrage/show/History.py [hunk 4]
```diff
@@ -57,32 +57,55 @@ def get(self, limit=100, action=None):
...
         else:
-            if actions:
-                results = self.db.select(common_sql + filter_sql + order_sql + 'LIMIT ?', actions + [limit])
-            else:
-                results = self.db.select(common_sql + order_sql + 'LIMIT ?', [limit])
+            results = self.db.select(common_sql + order_sql)
 
         data = []
-        for result in results:
-            data.append({
```
Comment (fernandog) [inline sickrage/show/History.py]: @p0psicles can you revert this and try again?

File: sickrage/show/History.py [hunk 5]
```diff
@@ -57,32 +57,55 @@ def get(self, limit=100, action=None):
...
+                'provider': row['provider'],
+                'resource': row['resource'],
+                'time': row['date']
+            }
+            if not any((history['show_id'] == row['show_id'] and
+                        history['season'] == row['season'] and
+                        history['episode'] == row['episode'] and
+                        history['quality'] == row['quality']) for history in compact):
+                history = {
+                    'actions': [action],
```
Comment (p0psicles) [inline sickrage/show/History.py]: For each row your moving [action] into the key 'actions'. You'l have to make gui changes of your going this route.

Comment (fernandog) [inline sickrage/show/History.py]: @p0psicles 
this is the exact code from webserve. I just moved from webserve.py > history.py

mine works ok. I set to show 1000 releases and no issue at all

Comment (p0psicles) [inline sickrage/show/History.py]: Hmm strange. I'll try on fresh install lttr.

Comment (fernandog) [inline sickrage/show/History.py]: I got this error two. needed to switch do "detailed" view

File: contrib/nzbToMedia [hunk 6]
```diff
@@ -1 +1 @@
-Subproject commit c50957d0b4683a125df039329d8f90df630d4e28
+Subproject commit 5c37f4a79eaff3837182121f5c589e061ca404cf
```
Comment (medariox) [inline contrib/nzbToMedia]: Please keep this at commit c50957d0b4683a125df039329d8f90df630d4e28.

Comment (fernandog) [inline contrib/nzbToMedia]: @medariox i didn't change this. how to revert??

File: sickrage/show/History.py [hunk 7]
```diff
@@ -97,17 +97,17 @@ def get(self, limit=100, action=None):
...
                          if item['show_id'] == row['show_id'] and
                          item['season'] == row['season'] and
                          item['episode'] == row['episode'] and
                          item['quality'] == row['quality']][0]
-                history = compact[index]
+                history = compact_view[index]
                 history['actions'].append(action)
                 history['actions'].sort(key=lambda x: x['time'], reverse=True)
-        return compact[:limit]
+        return detailed_view[:limit], compact_view[:limit]
```
Comment (labrys) [inline sickrage/show/History.py]: avoid returning plain tuple, return a named tuple instead.

File: sickrage/show/History.py [hunk 8]
```diff
@@ -97,17 +98,18 @@ def get(self, limit=100, action=None):
...
                          if item['show_id'] == row['show_id'] and
                          item['season'] == row['season'] and
                          item['episode'] == row['episode'] and
                          item['quality'] == row['quality']][0]
-                history = compact[index]
+                history = compact_view[index]
                 history['actions'].append(action)
                 history['actions'].sort(key=lambda x: x['time'], reverse=True)
-        return compact[:limit]
+        history = namedtuple('history', ['detailed', 'compact'])
```
Comment (labrys) [inline sickrage/show/History.py]: history here is a class so should be capitalized.

**Comment by fernandog** [comment]
tested sick ShowsRage and nzb360. all ok

File: sickrage/show/History.py [hunk 9]
```diff
@@ -82,7 +78,38 @@ def get(self, limit=100, action=None):
...
+                        history['quality'] == row['quality']) for history in compact_view):
+                history = {
+                    'actions': [action],
+                    'episode': row['episode'],
+                    'quality': row['quality'],
+                    'resource': row['resource'],
+                    'season': row['season'],
+                    'show_id': row['show_id'],
+                    'show_name': row['show_name']
+                }
```
Comment (labrys) [inline sickrage/show/History.py]: we already know most, if not all, of history == row, so why reassign them?

**Comment by fernandog** [comment]
@labrys need to squash&merge history_fix_nt into history_fix

**Comment by fernandog** [comment]
@labrys can we move on ?  7 says open

**Comment by labrys** [comment]
@fernandog We can close the other but I hadn't merged it to your branch because my last push was only partially done, i just pushed it so that i could work on it from home.  Now that they're merged there's no reason to keep the other open though.

**Comment by fernandog** [comment]
closing my PR as history seems will be full rewriten

**Pull request closed without merging**